### PR TITLE
Feature/app 4650 issues ocean

### PR DESCRIPTION
**Pull request by AcassioM**
## Description

-> Adjusts in RadioButton
-> Adjusts int CheckBox
-> Adjusts in Badge
-> Adjusts in Tag
-> New component Alert in SwiftUI
-> New component Link in SwiftUI

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Screenshots (if appropriate):

https://github.com/ocean-ds/ocean-ios/assets/94627996/f25ade9b-3388-4805-83af-753080cfca06

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Components SwiftUI/Alert/OceanSwiftUI+Alert.swift [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        public lazy var uiView = self.hostingController.getUIView()
+        
+        // MARK: Builder
+        
+        public typealias Builder = (Alert) -> Void
+        
+        // MARK: Properties
+        
+        @ObservedObject public var parameters: AlertParameters
+        private var withIcon: CGFloat = 24
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/Alert/OceanSwiftUI+Alert.swift:68]: Ajustar:

O correto seria widthIcon.

Sugestão:

iconWidth, iconHeight

File: OceanComponents/Classes/Components SwiftUI/Link/OceanSwiftUI+Link.swift [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            if self.parameters.isEnabled {
+                defaultView
+            } else {
+                disabledView
+            }
+        }
+        
+        // MARK: Methods private
+        
+        private func getForengroundColor() -> Color {
```
Comment (renanmassaroto) [inline OceanComponents/Classes/Components SwiftUI/Link/OceanSwiftUI+Link.swift:192]: Ajustar:

O texto tá errado, o correto é getForegroundColor

**Review by renanmassaroto — APPROVED** [review]

File: OceanComponents/Classes/Components SwiftUI/Alert/OceanSwiftUI+Alert.swift [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        
+        private var textView: some View {
+            OceanSwiftUI.Typography.caption { label in
+                label.parameters.text = parameters.text
+            }
+        }
+        
+        private var withAction: some View {
+            Group {
+                if ((parameters.actionText?.isEmpty) != nil) {
```
Comment (vromeiro) [inline OceanComponents/Classes/Components SwiftUI/Alert/OceanSwiftUI+Alert.swift:103]: muda para if let actionText = parameters.actionText

File: OceanDesignSystem/Controllers/Components SwiftUI/ComponentsSwiftUIViewController.swift [hunk 4]
```diff
@@ -40,8 +40,14 @@ class ComponentsSwiftUIViewController: UITableViewController {
         tableView.deselectRow(at: indexPath, animated: true)
 
         switch self.designSystemComponentsTypeSelected! {
+        case .Alert:
+            self.modalPresentationStyle = .fullScreen
```
Comment (vromeiro) [inline OceanDesignSystem/Controllers/Components SwiftUI/ComponentsSwiftUIViewController.swift:44]: precisa do .fullScreen?

**Review by vromeiro — APPROVED** [review]

File: OceanDesignSystem/Controllers/Components SwiftUI/ComponentsSwiftUIViewController.swift (hunk 4 above)
Comment (AcassioM) [reply OceanDesignSystem/Controllers/Components SwiftUI/ComponentsSwiftUIViewController.swift:44]: Nao nao... esse eu estava testando e esqueci de remover. 😅

**Comment by AcassioM** [comment]
Só para registrar... o merge ficou bloqueado por causa de codigos repetidos na viewController de exemplo...

image

**Pull request merged**